compiler/rustc_span/src/hygiene.rs

```diff
@@ -264,7 +264,14 @@ impl ExpnId {
         HygieneData::with(|data| data.expn_data(self).clone())
     }
 
+    #[inline]
     pub fn is_descendant_of(self, ancestor: ExpnId) -> bool {
+        if ancestor.krate != self.krate {
```

Comment:
This is not exact. The current code merges all cases of `expn_id.index == 0` together as `ExpnId::root()`. It would be correct to add just above:
```
if ancestor == ExpnId::root() {
    return true;
}
```

Reply:
Got it, but do you mind explaining further? What is `index == 0`?

Reply:
>What does `something.index == 0` mean?

It means that `something` is a root expansion (code not from a macro).